### PR TITLE
Disable context menu action on selection of multiple entries

### PR DESCRIPTION
**Pull request by motokito**
If multiple entries are selected, some  context menu, toolbar and menu actions are now disabled.
Disabled actions:
- Print entry preview
- Send as email
- Attach file
- Edit entry
- Open folder (enable only, if file exist)
- Open file (enable only, if file exist)
- Open URL or DOI (enable only, if url or doi exist)
- Merge entries (enable only on select of two entries)

These actions make no sense to be used when there are multiple entries selected.
For further information have a look at #463 

Context menu before change:

2016-09-22_15h50_28

Context menu after change:

2016-09-22_15h50_45

Edit menu before change:

2016-09-22_15h51_28

Edit menu after change:

2016-09-22_15h51_55

Bibtex menu before change:

2016-09-22_15h52_21

Bibtex menu after change:

2016-09-22_15h52_44

Quality menu before change:

2016-09-22_16h34_33

Quality menu after change:

2016-09-22_16h34_52

Tools menu before change:

2016-09-22_16h35_17

Tools menu after change:

2016-09-22_18h08_02

Toolbare button "Edit Entry" before change:

2016-09-22_15h53_17

Toolbare button "Edit Entry" after change:

2016-09-22_15h53_44

Toolbare button "Merge entries" before change:

2016-09-22_16h38_10

Toolbare button "Merge entries" after change:

2016-09-22_16h37_00

Please take a look for merge. THX 😄 

- [x] Change in CHANGELOG.md described
- [x] Screenshots added (for bigger UI changes)
- [x] Manually tested changed features in running JabRef
- [ ] Internal QS

### CODE REVIEW

**Comment by motokito** [comment]
Complete to disable menu item, toolbar button and context menu on multiple select of entries

Context menu before change:

2016-09-22_15h50_28

Context menu after change:

2016-09-22_15h50_45

Edit menu before change:

2016-09-22_15h51_28

Edit menu after change:

2016-09-22_15h51_55

Bibtex menu before change:

2016-09-22_15h52_21

Bibtex menu after change:

2016-09-22_15h52_44

Toolbare button "Edit Entry" before change:

2016-09-22_15h53_17

Toolbare button "Edit Entry" after change:

2016-09-22_15h53_44

Please take a look for merge. THX 😄

**Comment by boceckts** [comment]
Merge entries needs to be enabled only if exactly two entries are selected, also disable the following if multiple entries are selected:
Tools -> open file
Tools -> Open URL or DOI

File: CHANGELOG.md [hunk 1]
```diff
@@ -42,6 +42,7 @@ We refer to [GitHub issues](https://github.com/JabRef/jabref/issues) by using `#
 - Importer for MODS format added
 - [#2012](https://github.com/JabRef/jabref/issues/2012) Implemented integrity check for `month` field: Should be an integer or normalized (BibLaTeX), Should be normalized (BibTeX)
 - [#1779](https://github.com/JabRef/jabref/issues/1779) Implemented integrity check for `bibtexkey` field: Empty bibtexkey
+- [#463](https://github.com/JabRef/jabref/issues/463): Disable menu-item, toolbar-buttons and context menu-item while multiple entries are selected
```
Comment (boceckts) [inline CHANGELOG.md]: Reword to "Disable some menu items, toolbar buttons and context menu items...."

Comment (motokito) [inline CHANGELOG.md]: Done 😄

**Comment by motokito** [comment]
All things are complete 😄

**Comment by boceckts** [comment]
Please resolve conflicts.

**Comment by motokito** [comment]
Done 👍

File: CHANGELOG.md (hunk 1 above)
Comment (koppor) [inline CHANGELOG.md]: "some" -> "certain"

File: src/main/java/net/sf/jabref/gui/JabRefFrame.java [hunk 2]
```diff
@@ -1885,6 +1910,36 @@ public void setProgressBarMaximum(final int value) {
 
     }
 
+    /**
+     * Return a boolean, if the selected entry have file
+     * @param selectEntryList A selected entries list of the current base pane
+     * @return true, if the selected entry contains file
+     * <p/>
```
Comment (koppor) [inline src/main/java/net/sf/jabref/gui/JabRefFrame.java]: Why the `<p/>` sign? Please remove. Also applies for the other comments.

File: src/main/java/net/sf/jabref/gui/JabRefFrame.java [hunk 3]
```diff
@@ -1583,6 +1600,14 @@ public void updateEnabledState() {
...
             setEnabled(sharedDatabaseOnlyActions, isShared);
             setEnabled(noSharedDatabaseActions, !isShared);
+
+            boolean oneEntrySelected = current.getSelectedEntries().size() == 1;
+            setEnabled(oneEntryOnlyActions, oneEntrySelected);
+            setEnabled(oneEntryWithFileOnlyActions, isExistFile(current.getSelectedEntries()));
+            setEnabled(oneEntryWithURLorDOIOnlyActions, isExistURLorDOI(current.getSelectedEntries()));
+
+            boolean twoEntrySelected = current.getSelectedEntries().size() == 2;
+            setEnabled(twoEntryOnlyActions, twoEntrySelected);
```
Comment (koppor) [inline src/main/java/net/sf/jabref/gui/JabRefFrame.java]: It should be `twoEntries`

**Comment by koppor** [comment]
Besides minor comments: LGTM. - The send-as-email-action internally works with multiple entries, but Outlook cannot receive that long string as text. Thus, it is OK for me to exclude this action.

File: src/main/java/net/sf/jabref/gui/JabRefFrame.java (hunk 3 above)
Comment (motokito) [inline src/main/java/net/sf/jabref/gui/JabRefFrame.java]: Done 👍

File: src/main/java/net/sf/jabref/gui/JabRefFrame.java (hunk 2 above)
Comment (motokito) [inline src/main/java/net/sf/jabref/gui/JabRefFrame.java]: Done 👍

**Comment by motokito** [comment]
All things are done 🎅

**Comment by Siedlerchr** [comment]
Okay, looks good now. Merge it in. :+1:

**Pull request merged**